doc/contributor/howto-guide-setup-development-workstation.md

```diff
@@ -123,6 +123,42 @@ alternatives.
 ./ci/install-cloud-sdk.sh
 ```
 
+If you use an editor that uses `compile_commands.json` for autocomplete, there
+is a straight-forward way to generate it using Bazel. First, install Bazelisk:
+
+```console
+go get github.com/bazelbuild/bazelisk
+# Alternatively, if you have Go 1.17 or later
+go install github.com/bazelbuild/bazelisk@latest
+```
+
+Add Go binaries to your PATH:
+
+```console
+export PATH=$PATH:$(go env GOPATH)/bin
+```
+
+Name this script `bazel-compile-commands.sh`, add it to `~/.local/bin` and
+`chmod +x` it:
+
+```console
+#!/bin/bash
```

Comment:
This seems more complicated than this:

https://cmake.org/cmake/help/latest/variable/CMAKE_EXPORT_COMPILE_COMMANDS.html

Is it because installing the deps is easier with Bazel vs. CMake?  Maybe we should say something like:

If you are using CMake, read about CMAKE_EXPORT_COMPILE_COMMANDS (<<<< link) to generate a compilation database.  If you prefer using Bazel, the following steps will generate one for you.

WDYT?

Reply:
Could be wrong, but there currently isn't any way to pass extra Cmake flags to the docker build? Alternatively, one could have a separate CMake build just for `compile_commands.json` but that got unwieldly. So I looked into using ninja directly, but also ran into issues with that. Even though the Bazel approach looks like it requires more steps on the surface I found it to actually be the most straightforward overall. But this is based on fuzzy memory and trying to figure out the cleanest way of doing this for months. I will add the link either way.

Reply:
I remember now. There were weird issues with file paths when trying to generate the file within Docker. For some reason, the Bazel approach does not have that issue.

Reply:
Ah, yes, there would be issues with the paths, as inside Docker we mount the source and $HOME in weird directories.  I think this looks fine.  Let me summon @devjgm because he uses Bazel for development and may want to chime in.

Reply:
LGTM